projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6517,10 +6518,18 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
             return;
         }
         if (commit) {
+            this.verticalScrollContainer.onDataChanged.pipe(first()).subscribe(() => {
+            // A check whether the row is in the current view
+            const shouldScroll = this.navigation.shouldPerformVerticalScroll(this.dataLength - 1, 0);
```

Comment:
New record will not necessarily be added to the end of the data view. Should check its resolved index after the pipes.